### DISCUSSION TITLE
Mapping between taxonomy and billing codes

### DISCUSSION THREAD

**Discussion by r-shelby** [CMSgov/price-transparency-guide#610] [Q&A]
Has anyone found a standard source to map provider taxonomy / specialty to a set of billing codes? Per the requirement, most of the In-Network MRFs published include every billing code for every NPI, which leads to billions of rows of disposable data. Looking for a way to map active provider NPI to billing codes.

**Comment by TiaGossSawhney** [comment]
There is no standard way summarize the data, but there is a good do-it-yourself way.  The data prep takes work, so you might want to find someone who has steps 1 through 3 completed and who can guide and review steps 4 and 5.

1)  Map provider NPI to primary taxonomy code.  To do this you need to download the NPI registry from here: 
 https://npiregistry.cms.hhs.gov/search.  Please note that the primary taxonomy code is not necessarily taxonomy code 1.  

2)  Map primary taxonomy code to specialty.  You can start with the crosswalk table from here: https://www.nucc.org/index.php/code-sets-mainmenu-41/provider-taxonomy-mainmenu-40.

3)  Create a list of the top HCPCS codes (CPT + Level 2 HCPCS) for each specialty , where a top code is defined as total Medicare allowed amount (not units of service).  You can do so using your own plan data.  Or you can use Medicare public data found here:  https://www.cms.gov/OpenPayments/Data.  The downside for using Medicare data if you are looking at fee schedules for commercial plans is it does not contain pediatric data and has very little obstetric data.  For most specialties, however, it's a decent representation of the mix of the commercial services provided by the specialty.  (Example, internal medicine physicians mostly provide office visits, irrespective of whether the patient is covered by Medicare or commercial insurance.)

4)  Create abbreviated fee schedules by pulling only the top HCPCS codes from the MRFs.

5)  If you want to compare one fee schedule to another, calculate the weighted average fee using the number of units or distribution of units (not allowed amounts) from step 3.
